### PR TITLE
[FLINK-32559] Deprecate Queryable State

### PR DESCRIPTION
**Pull request by xintongsong**

### CODE REVIEW

File: docs/content.zh/docs/deployment/resource-providers/standalone/docker.md [hunk 1]
```diff
@@ -574,7 +574,7 @@ You can customize the Flink image in several ways:
 
     $ echo "
     # enable an optional library
-    ln -fs /opt/flink/opt/flink-queryable-state-runtime-*.jar /opt/flink/lib/
+    ln -fs /opt/flink/opt/flink-sql-gateway-*.jar /opt/flink/lib/
```
Comment (masteryhx) [inline docs/content.zh/docs/deployment/resource-providers/standalone/docker.md:577]: Why the sql-gateway jar is needed ?

**Review by masteryhx — CHANGES_REQUESTED** [review]
Thanks for the pr!
I think we may also need to:
1. Remove the generated doc: queryable_state_configuration.html
2. Remove other related doc/link, e.g.: 
- security-ssl.md (###Queryable State)
- stateful-stream-processing.md ("Queryable state allows you to access state from outside of Flink during runtime.")
- etl.md ("queryable: Flink state can be queried externally via the Queryable State API")
- overview.md (###Queryable State)

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/KeyedStream.java [hunk 2]
```diff
@@ -1055,8 +1055,11 @@ protected SingleOutputStreamOperator<T> aggregate(AggregationFunction<T> aggrega
      *
      * @param queryableStateName Name under which to the publish the queryable state instance
      * @return Queryable state instance
+     * @deprecated The Queryable State feature is deprecated since Flink 1.18, and will be removed
+     *     in a future Flink major version.
      */
     @PublicEvolving
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (xintongsong) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/KeyedStream.java:1062]: There's another PR working on deprecating all Scala APIs.
https://github.com/apache/flink/pull/23026

File: docs/content.zh/docs/deployment/resource-providers/standalone/docker.md (hunk 1 above)
Comment (xintongsong) [reply docs/content.zh/docs/deployment/resource-providers/standalone/docker.md:577]: I think here the purpose is to show how to enable an optional library with docker. It doesn't matter which library is brought in. Thus, if we don't want queryable state to appear in the documentation, any other library should be fine.

**Comment by xintongsong** [comment]
> 1. Remove the generated doc: queryable_state_configuration.html
> 2. Remove other related doc/link, e.g.:

Yep. I was hoping the CI docs_404_check to find all the missing links.

**Comment by xintongsong** [comment]
@masteryhx Comments addressed, PTAL.

**Review by masteryhx — APPROVED** [review]
Thanks for the quick update, LGTM.

**Review by wangyang0918 — APPROVED** [review]
I have no more comments. +1

**Comment by xintongsong** [comment]
Thanks for the review. I'll merge this once it passes the CI.

**Pull request closed without merging**